app/controllers/concerns/authentication/controller.rb

```diff
@@ -21,153 +20,152 @@ module Controller
 
     protected
 
-      # Verifies the current_user. The user is considered verified if one or more
-      # criteria are met. If not, a redirect is performed.
-      #
-      # Criteria:
-      #
-      #  :user       - Checks that current_user matches the given user or :any
-      #  :admin      - Checks that current_user is an admin
-      #  :moderator  - Checks that current_user is a moderator
-      #  :user_admin - Checks that current_user is a user admin
-      #
-      # Other options:
-      #
-      #  :notice   - Notice to display if verification fails
-      #  :redirect - URL to redirect to if verification fails
-      #
-      # Examples:
-      #
-      #  # Require any user
-      #  verify_user(user: :any, redirect: login_users_url, notice: 'You must be logged in!')
-      #
-      #  # Only accessible by a moderator
-      #  verify_user(moderator: true, notice: 'You must be a moderator!')
-      #
-      #  # Only accessible by a user admin or the user who owns the invite
-      #  verify_user(user: @invite.user, user_admin: true)
-      #
-      def verify_user(options={})
-        options = default_verify_user_options(options)
-
-        verified = false
-        if current_user?
-          verified ||= options[:user] == :any         if options[:user]
-          verified ||= options[:user] == current_user if options[:user]
-          verified ||= current_user.admin?            if options[:admin]
-          verified ||= current_user.moderator?        if options[:moderator]
-          verified ||= current_user.user_admin?       if options[:user_admin]
-        end
-
-        handle_unverified_user(options) unless verified
-        return verified
-      end
+    # Verifies the current_user. The user is considered verified if one or more
+    # criteria are met. If not, a redirect is performed.
+    #
+    # Criteria:
+    #
+    #  :user       - Checks that current_user matches the given user or :any
+    #  :admin      - Checks that current_user is an admin
+    #  :moderator  - Checks that current_user is a moderator
+    #  :user_admin - Checks that current_user is a user admin
+    #
+    # Other options:
+    #
+    #  :notice   - Notice to display if verification fails
+    #  :redirect - URL to redirect to if verification fails
+    #
+    # Examples:
+    #
+    #  # Require any user
+    #  verify_user(user: :any, redirect: login_users_url, notice: 'You must be logged in!')
+    #
+    #  # Only accessible by a moderator
+    #  verify_user(moderator: true, notice: 'You must be a moderator!')
+    #
+    #  # Only accessible by a user admin or the user who owns the invite
+    #  verify_user(user: @invite.user, user_admin: true)
+    #
+    def verify_user(options={})
```

Comment:
Assignment Branch Condition size for verify_user is too high. [23.54/15]<br>Perceived complexity for verify_user is too high. [8/7]<br>Surrounding space missing in default value assignment.